### PR TITLE
[WIP] Implement WebGL depth texture

### PR DESCRIPTION
**Pull request by ivanshen**
As per https://github.com/servo/servo/issues/22796
---

- [ ] `./mach build -d` does not report any errors
- [ ] `./mach test-tidy` does not report any errors
- [ ] These changes fix #___ (GitHub issue number if applicable)

- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because ___

### CODE REVIEW

**Comment by ivanshen** [comment]
@jdm I'm unsure where the implementation for the [error conditions from depth texture](https://www.khronos.org/registry/webgl/extensions/WEBGL_depth_texture/) go. This is what I have so far. Would you be able to give me feedback on what I have?

File: components/script/dom/webglframebuffer.rs [hunk 1]
```diff
@@ -571,7 +572,7 @@ impl WebGLFramebuffer {
         self.upcast::<WebGLObject>()
             .context()
             .send_command(WebGLCommand::FramebufferTexture2D(
-                constants::FRAMEBUFFER,
+                gl::UNSIGNED_INT_24_8,
```
Comment (jdm) [inline components/script/dom/webglframebuffer.rs:575]: This change does not make sense to me. What is the intention here?

File: components/script/dom/webglrenderingcontext.rs [hunk 2]
```diff
@@ -4227,7 +4228,7 @@ impl WebGLRenderingContextMethods for WebGLRenderingContext {
             handle_potential_webgl_error!(self, self.validate_ownership(texture), return);
         }
 
-        if target != constants::FRAMEBUFFER {
+        if target != gl::UNSIGNED_INT_24_8 {
```
Comment (jdm) [inline components/script/dom/webglrenderingcontext.rs:4231]: This also doesn't make sense. The spec says ` The framebufferTexture2D entry point is extended to accept the target parameter DEPTH_ATTACHMENT and DEPTH_STENCIL_ATTACHMENT`, which is not what's being checked.

**Review by jdm — CHANGES_REQUESTED** [review]
To answer your question about the error conditions, I would expect to see them in the TexImage2D and other methods. Almost all of them are checks against the parameters that are passed in.

**Comment by jdm** [comment]
Sorry for missing your request for feedback @ivanshen!

**Comment by jdm** [comment]
Closing due to inactivity.

**Pull request closed without merging**